src/main/rascal/lang/jimple/analysis/saa/DominanceFrontier.rsc

```diff
@@ -0,0 +1,33 @@
+module lang::jimple::analysis::saa::DominanceFrontier
+
+import Set;
+import analysis::graphs::Graph;
+import lang::jimple::analysis::FlowGraph;
+import lang::jimple::analysis::saa::DominanceTree;
+
+public map[Node, set[Node]] createDominanceFrontier(FlowGraph flowGraph, map[&T, set[&T]] dominanceTree) {
+	dominanceFrontiers = ();
+	
+	for(<origin, child> <- flowGraph) {
+		if(isJoinNode(flowGraph, child))
+			dominanceFrontiers = calculateDominanceFrontier(origin, child, dominanceFrontiers, dominanceTree);
+	};
+	
+	return dominanceFrontiers;
+}
+
+public map[Node, set[Node]] calculateDominanceFrontier(Node origin, Node destination, map[Node, set[Node]] dominanceFrontier, map[&T, set[&T]] dominanceTree) {
+	temp = origin;
+	
+	while(temp != findIdom(dominanceTree, destination)) {
+		originDominanceFrontierValue = if(dominanceFrontier[temp]?) dominanceFrontier[temp]; else {};
+		dominanceFrontier[temp] = originDominanceFrontierValue + {destination};
+		temp = findIdom(dominanceTree, temp);
+	};
+
+	return dominanceFrontier;
+}
```

Comment:
Implementation of:

![Screen Shot 2020-10-21 at 21 01 07](https://user-images.githubusercontent.com/14284674/96802586-8c288580-13e0-11eb-82ef-f8b3684d741b.png)